daemon/monitor.go

```diff
@@ -39,10 +39,15 @@ func (daemon *Daemon) ProcessEvent(id string, e libcontainerd.EventType, ei libc
 		if runtime.GOOS == "windows" {
 			return errors.New("received StateOOM from libcontainerd on Windows. This should never happen")
 		}
+
+		c.Lock()
 		daemon.updateHealthMonitor(c)
 		if err := c.CheckpointTo(daemon.containersReplica); err != nil {
+			c.Unlock()
```

Comment:
if the function calls above panic the lock won't be released. Use deferred closure or refactor in a function?

Reply:
@cpuguy83 ^^

Reply:
Updated